test/jdk/javax/swing/text/html/HTMLDocument/HTMLTextDecoration.java

```diff
@@ -0,0 +1,157 @@
+            <p style='text-decoration: underline'><strike>underline + line-through?</strike></p>
+
+            <p style='text-decoration: line-through'><span style='text-decoration: underline'>underline + line-through?</span></p>
+            <p style='text-decoration: underline'><span style='text-decoration: line-through'>underline + line-through?</span></p>
+
+            <p class="underline"><span class="lineThrough">underline + line-through?</span></p>
+            <p class="underline"><s>underline + line-through?</s></p>
+            <p class="underline"><strike>underline + line-through?</strike></p>
+
+            <p class="lineThrough"><span class="underline">underline + line-through?</span></p>
+            <p class="lineThrough"><u>underline + line-through?</u></p>
+
+            <div class="underline"><span class="lineThrough">underline + line-through?</span></div>
+            <div class="underline"><s>underline + line-through?</s></div>
+            <div class="underline"><strike>underline + line-through?</strike></div>
+
+            <div class="lineThrough"><span class="underline">underline + line-through?</span></div>
+            <div class="lineThrough"><u>underline + line-through?</u></div>
+
+            <div class="underline"><p class="lineThrough">underline + line-through?</p></div>
+            <div class="lineThrough"><p class="underline">underline + line-through?</p></div>
+
+            <div class="underline"><div class="lineThrough">underline + line-through?</div></div>
+            <div class="lineThrough"><div class="underline">underline + line-through?</div></div>
+            </body>
+            </html>
+            """;
+
+    public static void main(String[] args) {
+        final JEditorPane html = new JEditorPane("text/html", HTML);
+        html.setEditable(false);
+
+        final Dimension size = html.getPreferredSize();
+        html.setSize(size);
+
+        BufferedImage image = new BufferedImage(size.width, size.height,
+                                                BufferedImage.TYPE_INT_RGB);
+        Graphics g = image.createGraphics();
+        // Paint the editor pane to ensure all views are created
+        html.paint(g);
+        g.dispose();
+
+        int errorCount = 0;
+        String firstError = null;
+
+        System.out.println("----- Views -----");
+        final View bodyView = html.getUI()
+                                  .getRootView(html)
+                                  .getView(1)
+                                  .getView(1);
+        for (int i = 0; i < bodyView.getViewCount(); i++) {
+            View pView = bodyView.getView(i);
+            View contentView = getContentView(pView);
+
+            String decoration =
+                    contentView.getAttributes()
+                               .getAttribute(CSS.Attribute.TEXT_DECORATION)
+                               .toString();
```

Comment:
It might be good to add a null check before calling .toString() since `contentView.getAttributes().getAttribute(CSS.Attribute.TEXT_DECORATION)` can return null.

Reply:
No, it can't. In the test, all the views that end up in `contentView` must have the `CSS.Attribute.TEXT_DECORATION` attribute and its value must contain both `underline` and `line-through`. It is what the test verifies.

If the test is modified and `contentView` does not have the `CSS.Attribute.TEXT_DECORATION` attribute, the test will fail with `NullPointerException` which is fine.